### ISSUE TITLE
Animation missing error

### ISSUE THREAD

**Issue by arobinson** [Dugy/Legend_of_the_Invincibles#291]
![image](https://user-images.githubusercontent.com/93213/97215988-c9976500-178a-11eb-980d-786ef35d5a66.png)

[LotI2-Across the Barren Land-Auto-Save48.gz](https://github.com/Dugy/Legend_of_the_Invincibles/files/5440807/LotI2-Across.the.Barren.Land-Auto-Save48.gz)

To reproduce advance Miry to a chaos rider in the cave

**Comment by dalong76** [comment]
I believe the cause is that the item Overdrawer gives you a bonus attack which is a clone of the original crossbow attack. Since the chaos rider has no crossbow attack, there is no attack to clone, thus no animation to clone, and thus you get that error. Not easy to fix.

**Comment by Dugy** [comment]
Yeah, I didn't think about leftover items when designing the bonus attacks. There are basically two solution to this:
1. If no attack to derive the bonus from is found, the bonus attack does not appear
2. Add a basic attack animation to every unit that can have leftover items
The first option is far easier to implement. The second option would be much more interesting from the point of allowing players to create interesting units, however, it leads to additional problems like liches (and everything they can advance to) that can have leftover items from pretty much any type due to the `necromancy` ability.

What do you guys think about it?

**Comment by dalong76** [comment]
I don't think your second is worth the effort.
A third possibility is to create a generic animation for the bonus attack. In this case, the generic animation would be just the missile. The unit doesn't move, but the missile flies, so there is animation. It isn't perfect, but it is far easier. A melee attack could be just a generic weapon swinging around.
